R/shared-methods.R

```diff
@@ -102,28 +102,38 @@ DataClass <- R6::R6Class(
     verbose = NULL,
     #' @field steps Boolean. Keep data from each processing step.
     steps = NULL,
+    #' @field region_codes string or tibble Region codes for the target country
+    region_codes = NULL,
+
+    #' @description Place holder for custom country specific function to load
+    #' region codes.
+    set_region_codes = function() {
+    },
 
     #' @description General function for getting region codes for given region.
     #' @rdname get_region_codes
     #' @importFrom rlang .data
     get_region_codes = function() {
       tar_level <- paste0("level_", self$level, "_region")
       tar_level_name <- self[[tar_level]]
-      codes <- covidregionaldata::region_codes %>%
-        filter(
-          .data$country %in% self$country,
-          .data$level %in% tar_level
-        )
+      self$data <- list(country = self$country, level = tar_level_name)
       message_verbose(
         self$verbose, "Processing data for ",
         self$country, " by ", tar_level_name
       )
-
-      self$data <- list(country = self$country, level = tar_level_name)
-
-      if (nrow(codes) == 1) {
-        self$data$code <- codes$name[[1]]
-        self$data$codes_lookup <- codes$codes[[1]]
+      code_name <- self$set_region_codes()
+      if (is.null(self$region_codes)) {
```

Comment:
I don't think we need this function?

Reply:
I was surprised when I went looking (trying to figure out something else) the range of things which `get_region_codes` does. It seems very much an initializer (setting up `self$data` is pretty key). If it were given a different name and the region codes functionality were broken out somewhere else would this work better? (I tried calling `get_region_codes` twice with unfortunate consequences.)